Code/Engine/GameEngine/GameState/Implementation/GameState.cpp

```diff
@@ -132,33 +79,65 @@ void ezGameState::CreateActors()
   //  }
   //#endif
 
-  // TODO: VR support
-  // if (m_bVirtualRealityMode)
-  //{
-  //  wndDesc.m_bClipMouseCursor = false;
-  //  wndDesc.m_bShowMouseCursor = true;
-  //  wndDesc.m_WindowMode = ezWindowMode::WindowResizable;
-  //}
-
-  ezUniquePtr<ezActor> pActor = EZ_DEFAULT_NEW(ezActor, "Main Window", this);
-
-  ezUniquePtr<ezWindow> pMainWindow = CreateMainWindow();
-  EZ_ASSERT_DEV(pMainWindow != nullptr, "To change the main window creation behavior, override ezGameState::CreateActors().");
-
-  ezUniquePtr<ezWindowOutputTargetBase> pOutput = CreateMainOutputTarget(pMainWindow.Borrow());
-
-  ConfigureMainWindowInputDevices(pMainWindow.Borrow());
-
-  ezUniquePtr<ezActorPluginWindowOwner> pWindowPlugin = EZ_DEFAULT_NEW(ezActorPluginWindowOwner);
-
-  pWindowPlugin->m_pWindow = std::move(pMainWindow);
-  pWindowPlugin->m_pWindowOutputTarget = std::move(pOutput);
-
-  SetupMainView(pWindowPlugin->m_pWindowOutputTarget.Borrow(), pWindowPlugin->m_pWindow->GetClientAreaSize());
+  // Init XR
+  const ezXRConfig* pConfig = ezGameApplicationBase::GetGameApplicationBaseInstance()->GetPlatformProfile().GetTypeConfig<ezXRConfig>();
+  ezXRInterface* pXRInterface = nullptr;
+  if (pConfig && pConfig->m_bEnableXR)
+  {
+    if (ezXRInterface* pXR = ezSingletonRegistry::GetRequiredSingletonInstance<ezXRInterface>())
```

Comment:
I think the required version will assert that the interface exists.

Reply:
Nice find, I didn't knew there were two different functions.